### PR TITLE
Default containers

### PR DESCRIPTION
**Pull request by nicola**

### CODE REVIEW

**Comment by dmitrizagidulin** [comment]
Addresses https://github.com/linkeddata/ldnode/issues/190 and https://github.com/solid/solid/issues/23

**Comment by deiu** [comment]
Can we close this PR without merging, and create a separate one that only does the preferences file? This decision is motivated by https://github.com/solid/solid/issues/23 and [data discovery](https://github.com/solid/solid/blob/master/proposals/data-discovery.md).

**Comment by dmitrizagidulin** [comment]
I think we should, yeah. (close this PR, open a separate one, copy just the code we need)

**Comment by dmitrizagidulin** [comment]
@nicola ?

**Comment by nicola** [comment]
@dmitrizagidulin the code is the same except the json containing the default folders

**Comment by dmitrizagidulin** [comment]
@nicola so is that a yes or no? :) (to closing this PR, and re-adding the code in a new one?)

**Comment by nicola** [comment]
Isn't renaming this PR enough?

**Comment by deiu** [comment]
There's not much that can be reused from this particular PR, not even the name. Hence my suggestion to close it and start fresh.

File: lib/identity-provider.js [hunk 1]
```diff
@@ -19,6 +19,38 @@ var pki = forge.pki
...
+    type: 'PrivateWorkspace'
+  }, {
+    name: 'Work',
+    label: 'Work workspace'
+  }, {
+    name: 'Shared',
+    label: 'Shared workspace',
+    type: 'SharedWorkspace'
+  }, {
+    name: 'Preferences',
```
Comment (deiu) [inline lib/identity-provider.js]: Preferences -> Settings

File: lib/identity-provider.js [hunk 2]
```diff
@@ -19,6 +19,38 @@ var pki = forge.pki
 var parse = require('./utils').parse
 var stringToStream = require('./utils').stringToStream
```
Comment (deiu) [inline lib/identity-provider.js]: No longer required to create the "workspaces"

File: lib/identity-provider.js [hunk 3]
```diff
@@ -105,6 +139,57 @@ IdentityProvider.prototype.create = function (options, cert, callback) {
   })
 }
 
+function createPreferences (options) {
+  var graph = $rdf.graph()
+  var pref = options.url + 'Preferences/prefs.ttl'
```
Comment (deiu) [inline lib/identity-provider.js]: Preferences -> Settings

File: lib/identity-provider.js [hunk 4]
```diff
@@ -105,6 +139,57 @@ IdentityProvider.prototype.create = function (options, cert, callback) {
...
+
+function createWorkSpace (options, workspace) {
+  var resource = options.url + workspace.name + '/'
+  var acl = resource + options.suffixAcl
+
+  var aclGraph = createAcl(
+    acl,
+    resource,
+    options.agent,
+    options.email)
```
Comment (deiu) [inline lib/identity-provider.js]: No need to create the acls anymore for the workspaces. Except for Inbox which is append-only.

File: lib/identity-provider.js [hunk 5]
```diff
@@ -105,6 +139,57 @@ IdentityProvider.prototype.create = function (options, cert, callback) {
...
+
+  if (workspace.type === 'PublicWorkspace') {
```
Comment (deiu) [inline lib/identity-provider.js]: Optional

**Comment by deiu** [comment]
:+1: to merge

**Comment by dmitrizagidulin** [comment]
+1 from me (local tests pass).

**Comment by deiu** [comment]
-1 for me, see my comments [here is Nicola from @deiu's computer]

**Pull request merged**